core/whistle-1.0.0/src/wh_cache.erl

```diff
@@ -143,6 +143,19 @@ start_link(Name, ExpirePeriod, Props) ->
                                    )
     end.
 
+-spec build_bindings(wh_proplist()) -> wh_proplist().
+build_bindings([[]]) -> document_bindings([[]]);
+build_bindings(BindingProps) ->
+    [database_binding() | document_bindings(BindingProps)].
+
+-spec document_bindings(wh_proplist()) -> wh_proplist().
+document_bindings(BindingProps) ->
+     [{'conf', ['federate'|P]} || P <- BindingProps].
+
+-spec database_binding() -> tuple().
+database_binding() ->
+    {'conf', ['federate', {'type', <<"database">>}]}.
```

Comment:
this here `{'type', <<"database">>}.` or define a macro with this value

in the end we'll have `[{'conf', ['federate'|P]}`where P is document+(database if document not empty).